Spravo/AppDelegate.swift

```diff
@@ -45,4 +50,25 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 
     func applicationWillTerminate(_ application: UIApplication) {
     }
+    
+    @objc func checkForReachability(notification: Notification) {
+        guard let reachability = notification.object as? Reachability,
+            let rootVC = self.window?.rootViewController else { return }
+        if !reachability.isReachable() {
+            updateUIonMainThread {
+                guard let errorController = Storyboard.service.controller(withClass: ErrorScreenVC.self) else { return }
+                errorController.image = UIImage(named: "Disconnected")
+                errorController.text = NSLocalizedString("ImportPhoneContacts.ErrorInternetConnection", comment: "Request to check internet connection")
+                rootVC.modalPresentationStyle = .overCurrentContext
+                rootVC.present(errorController, animated: true, completion: nil)
```

Comment:
present will not always work, if there is a screen already presented (Alert for example)

Reply:
As for me :
guard let reachability = notification.object as? Reachability,
            let rootVC = self.window?.rootViewController else { return }
prevent error , about you talk, is not it?